### ISSUE TITLE
Unable to Sign In to Azure Storage Explorer on macOS - "Safari Cannot Find the Server" Error

### ISSUE THREAD

**Issue by SaraAlthubaiti** [microsoft/AzureStorageExplorer#8289]
Hello everyone,

I'm encountering an issue when trying to sign in to Azure Storage Explorer on my Mac. When I initiate the sign-in process, I'm directed to the sign-in page in my browser (I've tried both Safari and Chrome), and after entering my email and password, I receive an error stating, "Safari cannot find the server."

When I return to Azure Storage Explorer, I see the following error message:

```
{
  "message": "\"An HttpListenerException occurred while listening on http://localhost:54468/ for the system browser to complete the login. Possible cause and mitigation: the app is unable to listen on the specified URL; run 'netsh http add iplisten 127.0.0.1' from the Admin command prompt.\""
}
```

Details:

macOS Version: 13.7.1 (22H221)
Azure Storage Explorer Version: 1.36.2
Browsers Tried: Safari and Google Chrome
Troubleshooting Steps Taken:
Verified that my Mac is up to date.
Downloaded and installed the latest version of Azure Storage Explorer.
Tried using both Safari and Chrome to complete the sign-in.

Any guidance on resolving this for macOS would be greatly appreciated!

Thank you!

**Comment by JasonYeMSFT** [comment]
@SaraAlthubaiti Have you tried following the suggested step to allow listening to your localhost?

**Comment by SaraAlthubaiti** [comment]
@JasonYeMSFT The `netsh` command is a built-in tool in Windows, so I cannot try what's been suggested. I searched about equivalent command for Mac, I found this `sudo lsof -i :54468` but still nothing change and I'm getting the same error message.

**Comment by JasonYeMSFT** [comment]
@SaraAlthubaiti I see. Could you follow the steps to reproduce the issue and collect some verbose logs so hopefully we can see what the exact error is? https://learn.microsoft.com/en-us/troubleshoot/azure/azure-storage/blobs/alerts/storage-explorer-troubleshooting?tabs=Windows#authentication-logs

Are you running any webserver like program on your localhost that is already listening to localhost or 127.0.0.1?

**Comment by craxal** [comment]
Closing due to inactivity. If you are still encountering problems, please respond with the requested information.

**Comment by nyannios** [comment]
I am having this issue too on macOS Sonoma.

When I try to login on Storage Explorer, I get 

"Safari can’t open the page “http://localhost:50759/?error=access_denied&amp;error_subcode=cancel&amp;state=d82c984c-376c-4087-977c-b3b40e9a99d72edce30f-0559-4943-a6b7-59cf0ff18642”.

The error is: “Navigation failed because the request was for an HTTP URL with HTTPS-Only enabled” (WebKitErrorDomain:305

**Comment by craxal** [comment]
@nyannios Can you answer the questions asked previous by JasonYeMSFT, as well as provide the verbose logs?

**Comment by nyannios** [comment]
yes the version is 1.36.2 (103)   
  
on MacOS Sonoma   
  
  
  
  
  
  
  

  
  
>

**Comment by nyannios** [comment]
No, I do not have any web server software installed on the Mac OS system.  I do have various PowerShell modules.  I'll do a more thorough check what I have set up on that Mac when I get home .....

**Comment by craxal** [comment]
@nyannios Were you able to get more information from that check you mentioned?

**Comment by vadakr** [comment]
> I am having this issue too on macOS Sonoma.
> 
> When I try to login on Storage Explorer, I get
> 
> "Safari can’t open the page “[http://localhost:50759/?error=access_denied&amp;error_subcode=cancel&amp;state=d82c984c-376c-4087-977c-b3b40e9a99d72edce30f-0559-4943-a6b7-59cf0ff18642”](http://localhost:50759/?error=access_denied&amp;error_subcode=cancel&amp;state=d82c984c-376c-4087-977c-b3b40e9a99d72edce30f-0559-4943-a6b7-59cf0ff18642%E2%80%9D).
> 
> The error is: “Navigation failed because the request was for an HTTP URL with HTTPS-Only enabled” (WebKitErrorDomain:305


In Safari > Settings > Security, turn off the checkbox for 'Warn before connecting to a website over HTTP'

Try at your own risk because this will turn it off for all websites! It would be nice if Safari let us turn it off for 'localhost' or allowed us to add a list of excluded sites.
